src-java/blue-green/src/main/java/org/openkilda/bluegreen/ZkClient.java

```diff
@@ -93,12 +101,18 @@ protected void ensureZNode(byte[] value, String... path) throws KeeperException,
         }
     }
 
-    void validateNodes() throws KeeperException, InterruptedException {
+    void validateNodes() throws KeeperException, InterruptedException, IOException {
         ensureZNode(serviceName);
         ensureZNode(serviceName, id);
     }
 
+    /**
+     * Checks that connection to zookeeper is alive and nodes were validated.
+     */
     public boolean isConnectionAlive() {
+        if (zookeeper == null || !nodesValidated) {
```

Comment:
Why we have `!nodesValidated` here? Especially when `ZcClient` is not controlling this flag value?

Reply:
To be ensured that nodes were validated. In ZkWatchDog and ZkClient